src/Aspire.Hosting.Testing/DistributedApplicationTestingBuilder.cs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Aspire.Hosting.ApplicationModel;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Aspire.Hosting.Testing;
+
+/// <summary>
+/// Harness for running a distributed application for testing.
+/// </summary>
+/// <typeparam name="TEntryPoint">
+/// A type in the entry point assembly of the target Aspire AppHost. Typically, the Program class can be used.
+/// </typeparam>
+public sealed class DistributedApplicationTestingBuilder<TEntryPoint> : IDistributedApplicationBuilder where TEntryPoint : class
+{
+    private readonly SuspendingDistributedApplicationFactory _factory;
+    private readonly DistributedApplicationBuilder _applicationBuilder;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DistributedApplicationTestingBuilder{TEntryPoint}"/> class.
+    /// </summary>
+    public DistributedApplicationTestingBuilder() : this((_, __) => { })
+    { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DistributedApplicationTestingBuilder{TEntryPoint}"/> class.
+    /// </summary>
+    /// <param name="configureBuilder">The delegate used to configure the creation of the builder.</param>
+    public DistributedApplicationTestingBuilder(Action<DistributedApplicationOptions, HostApplicationBuilderSettings> configureBuilder)
+    {
+        _factory = new(configureBuilder);
+        _applicationBuilder = _factory.DistributedApplicationBuilder.Result;
```

Comment:
This just hang forever. Are the tests missing using the fixture or something?

Reply:
I introduced this issue when trying to make CI tests work. There is a fix PR: #2575